### PR TITLE
Add FFI for constructing fixed decimals from float

### PR DESCRIPTION
**Pull request by Manishearth**
Supersedes https://github.com/unicode-org/icu4x/pull/1477

Progress on #1472

### CODE REVIEW

File: ffi/diplomat/src/fixed_decimal.rs [hunk 1]
```diff
@@ -19,13 +21,67 @@ pub mod ffi {
...
         /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/fixed_decimal/decimal/struct.FixedDecimal.html) for more information.
         pub fn create(v: i32) -> Box<ICU4XFixedDecimal> {
             Box::new(ICU4XFixedDecimal(FixedDecimal::from(v)))
         }
 
+        /// Construct an [`ICU4XFixedDecimal`] from an float, with enough digits to recover
+        /// the original floating point in IEEE 754 without needing trailing zeros
+        ///
+        /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/fixed_decimal/decimal/struct.FixedDecimal.html#method.from_f64) for more information.
+        pub fn create_from_f64(f: f64) -> Option<Box<ICU4XFixedDecimal>> {
```
Comment (sffc) [inline ffi/diplomat/src/fixed_decimal.rs:45]: Nit: create_from_f64_with_max_precision

You know how I feel about making this have the appearance of being the default function. ;)

File: ffi/diplomat/src/fixed_decimal.rs [hunk 2]
```diff
@@ -19,13 +21,67 @@ pub mod ffi {
...
+            Some(Box::new(ICU4XFixedDecimal(
+                FixedDecimal::new_from_f64(f, DoublePrecision::Maximum).ok()?,
+            )))
+        }
+
+        /// Construct an [`ICU4XFixedDecimal`] from an float, with a given power of 10 for precision
+        ///
+        /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/fixed_decimal/decimal/struct.FixedDecimal.html#method.from_f64) for more information.
+        pub fn create_from_f64_with_precision(
```
Comment (sffc) [inline ffi/diplomat/src/fixed_decimal.rs:54]: Suggestion: _with_fixed_fraction_digits

File: ffi/diplomat/src/fixed_decimal.rs [hunk 3]
```diff
@@ -19,13 +21,67 @@ pub mod ffi {
...
+                    DoublePrecision::Magnitude(precision, rounding_mode.into()),
+                )
+                .ok()?,
+            )))
+        }
+
+        /// Construct an [`ICU4XFixedDecimal`] from an float, for a given number of digits
+        ///
+        /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/fixed_decimal/decimal/struct.FixedDecimal.html#method.from_f64) for more information.
+        pub fn create_from_f64_with_digits(
```
Comment (sffc) [inline ffi/diplomat/src/fixed_decimal.rs:71]: Suggestion: _with_fixed_precision

File: ffi/diplomat/src/fixed_decimal.rs (hunk 2 above)
Comment (Manishearth) [reply ffi/diplomat/src/fixed_decimal.rs:54]: That inverts the polarity of the argument and could be confusing?

**Review by sffc — APPROVED** [review]

**Pull request merged**